udsoncan/connections.py

```diff
@@ -735,15 +735,24 @@ def __init__(self, windll: str, rxid: int, txid: int, name: Optional[str] = None
         txFlags = TxStatusFlag.ISO15765_CAN_ID_29.value if txid >> 11 else TxStatusFlag.ISO15765_CAN_ID_11.value
 
         # Set up a J2534 interface using the DLL provided
-        self.interface = J2534(windll=windll, rxid=rxid, txid=txid, txFlags=txFlags)
+        try:
+            self.interface = J2534(windll=windll, rxid=rxid, txid=txid, txFlags=txFlags)
+        except FileNotFoundError:
+            raise RuntimeError('DLL not found')
 
         # Set the protocol to ISO15765, Baud rate to 500000
         self.protocol = Protocol_ID.ISO15765
         self.baudrate = 500000
         self.debug = debug
 
-        # Open the interface (connect to the DLL)
-        result, self.devID = self.interface.PassThruOpen()
+        try:
+            # Open the interface (connect to the DLL)
+            self.result, self.devID = self.interface.PassThruOpen()
+            self.log_last_operation("PassThruOpen", with_raise=True)
+        except WindowsError as e:

```

Comment:
How confident are you that these error code are exact and will always represent what you think they are?

Reply:
I used Windows documentation about this codes. Just google it.
And I tested connecting to the device when device already used by another thread.

Usally situation when application uses many threads underhood.